### ISSUE TITLE
As a user from another country I want to see names in a format that we use so that they reflect my cultural norms

### ISSUE THREAD

**Issue by euanmillar** [opencrvs/opencrvs-core#6830]
## Description

OpenCRVS is hardcoded to display names with family name appearing last in many places: {firstName} {middleName} {lastName}. 

But in many countries surname appears first so this should be configurable in record audit and other places for example.

This ticket describes 2 changes that we need to make.

## Tech tasks

**National System Admin – Create user form** 

![Image](https://github.com/user-attachments/assets/518801a8-2286-417c-975e-d25d211913a6)

1. When logged in as a national system administrator user you can create new users in offices.  The create new user form is not currently configurable and we have no plans to make it configurable.  However, as we often receive requests from countries to have the surname first in this form, we have decided to satisfy that requirement by putting surname first for ALL countries.  

We also want to remove the question that asks the user for a national ID number in this form as it is not required.

  - [x] Change name field ordering in user creation and edit flows to [Surname] [Firstname]. Rename the labels to be "User's surname" and "User's firstname".
  - [x] Remove the NID question from user creation / edit flow. Ensure possible NIDs are removed from `user-mgnt.users` collection. Write a migration to remove the field from existing users.

**How citizen data appears in the team user list, work queues and record audit**

![Image](https://github.com/user-attachments/assets/41423472-e8fa-4a20-bad6-d67179d47cbd)

![Image](https://github.com/user-attachments/assets/4cabd0d7-1dd2-4ab0-99b5-1ac38234a09d)

2. If we create a content management key called `constants.humanName` with a default value `{firstName} {lastName}`, then the client could read this content key and dynamically choose to represent names in work queues and record audit according to it.  If the content key was like this: `{lastName} {middleName} {firstName}` then the names could be rendered appropriately

  - [x] Create new copy item `constants.humanName` with a default value `{firstName} {lastName}` and start using it in the admin view's user listing. Keep in mind that this same constant is to be used for rendering record subject names so for instance a country might want to have this as `{lastName} {middleName} {firstName}`
- [x] Refactor `generateName` in packages/client/src/utils/data-formatting.ts so that it respects the name format defined in the team user list view

**Other applications**

  - [x] Go through the app and test for instance `{lastName} {middleName} {firstName}` properly applies everywhere such as work-queues and record audit.

- [x] In release notes: Add a few examples of different ways to configure this e.g. middle name doesn't exist OR <family name> <first name>

**Comment by euanmillar** [comment]
Decision needs to be made if name configuration is decoupled from the rest of registration form configuration.

**Comment by rikukissa** [comment]
@euanmillar ticket updated. Can be moved to **Ready to build**

**Comment by jpye-finch** [comment]
Could the user creation form is configurable with mandatory questions. So they change the order as they wish and add additional data

**Comment by rikukissa** [comment]
Updated task description to include configurable name formatting elsewhere in the app as Burkina Faso needs this

**Comment by euanmillar** [comment]
@rikukissa if we are doing this: "Ensure possible NIDs are removed from user-mgnt.users collection" wont we need a database migration?  I wonder if it is worth the effort.  It might be acceptable to just leave the national id prop in the database than maintain the migration file.  Just an idea

**Comment by rikukissa** [comment]
Good point, I'll update the task description. Long-term it's best if we aim to always have the database and data integrity in tip-top shape, so definitely worth writing a migration

**Comment by Erusha-Raju** [comment]
Milestone updated to 1.6.3 as agreed in the PM call 07/04
